src/Utility/Store.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * Generic class that expose all data manipulation methods.
+ * A Store can be constructed with any Key => Value structure (Map like object).
+ * It exposes all methods similar to aarray methods in one place.
+ *
+ * @author KhaaZ
+ *
+ * @template T
+ * @class Store
+ * @prop {Map} cache - Key => Value Data structure
+ */
+class Store {
+    /**
+     * Creates an instance of Store.
+     *
+     * @param {Map} cache - Key => Value data structure
+     * @memberof Store
+     */
+    constructor(cache) {
+        this.cache = cache || new Map();
+    }
+
+    // Direct action on cache
+
+    /**
+     * Current size of the cache
+     *
+     * @readonly
+     * @type {Number}
+     * @memberof Store
+     */
+    get size() {
+        return this.cache.size;
+    }
+
+    /**
+     * Iterator over cache.entries.
```

Comment:
```suggestion
     * Iterator over `cache.entries`
```
Just for JSDoc markdown